app/components/ChartAnnotations/index.js

```diff
@@ -15,12 +15,21 @@ class ChartAnnotations extends Component {
   }
 
   // Dragging an annotation - update dx/dy when drag ends
-  static onAnnotationDragEnd(data) {
-    ChartAnnotations.updateAnnotation(data);
+  static onAnnotationDragEnd(data, container) {
+    ChartAnnotations.updateAnnotation({
+      ...data,
+      ...ChartAnnotations.constrainAnnotation(data, container),
+    });
   }
 
   // Ensure annotations remain inside the chart container
   static constrainAnnotation(annotation, container) {
+    // Hide all but current annotation
+    d3v4.selectAll(`${container} .annotation`).each(function (data, index) {
```

Comment:
Why are you not using fat arrow function syntax?

Reply:
maintain 'this' for d3 - added comment.